src/fsharp/CompileOps.fs

```diff
@@ -5113,12 +5142,50 @@ module private ScriptPreprocessClosure =
             let tcConfigB = tcConfig.CloneOfOriginalBuilder 
             TcConfig.Create(tcConfigB, validate=false), nowarns
     
-    let FindClosureFiles(closureSources, tcConfig:TcConfig, codeContext, lexResourceManager:Lexhelp.LexResourceManager) =
-        let tcConfig = ref tcConfig
+    let FindClosureFiles(mainFile, _m, closureSources, origTcConfig:TcConfig, codeContext, lexResourceManager:Lexhelp.LexResourceManager) =
+        let tcConfig = ref origTcConfig
         
         let observedSources = Observed()
-        let rec loop (ClosureSource(filename, m, source, parseRequired)) = 
-          [     if not (observedSources.HaveSeen(filename)) then
+        let loadScripts = HashSet<_>()
+
+        // Resolve the packages
+        let rec resolveDependencyManagerSources scriptName =
+            if not (loadScripts.Contains scriptName) then
+                [ for kv in tcConfig.Value.packageManagerLines do
+                    let packageManagerKey,packageManagerLines = kv.Key,kv.Value
+                    match packageManagerLines with
+                    | [] -> ()
+                    | (_,m)::_ ->
+                        match origTcConfig.packageManagerLines |> Map.tryFind packageManagerKey with
+                        | Some oldDependencyManagerLines when oldDependencyManagerLines = packageManagerLines -> ()
+                        | _ ->
+                            match DependencyManagerIntegration.tryFindDependencyManagerByKey m packageManagerKey with
+                            | None ->
+                                errorR(DependencyManagerIntegration.createPackageManagerUnknownError packageManagerKey m)
```

Comment:
Perhaps this should just be a warning.  There will be a later error almost for sure - and if there isn't for some reason then it seems reasonable to proceed with compilation/execution

Reply:
Yes additionally, I think this would be a breaking change